generic-credential/main.go

```diff
@@ -163,11 +163,18 @@ func validateCredential(ctx context.Context, client *gptscript.GPTScript, tool s
 		return fmt.Errorf("error running tool: %w", err)
 	}
 
-	_, err = run.Text()
+	output, err := run.Text()
 	if err != nil {
 		errStr, _, _ := strings.Cut(err.Error(), ": exit status ")
 		return errors.New(errStr)
 	}
 
+	var errResp struct {
+		Error string `json:"error"`
+	}
+	if err := json.Unmarshal([]byte(output), &errResp); err == nil && errResp.Error != "" {
+		return errors.New(errResp.Error)
```

Comment:
Just to make sure: based on the changes in the other languages, I expect that `errResp.Error` is already JSON?

Reply:
It's normally just going to be a string. The `validationTool` tools, if they error, will print something in the format `{"error": "this is the error message"}` and then `exit(0)`, which is what we are looking for here.